### PR TITLE
[jsonrpc] Allow retrieving empty TV shows via VideoLibrary.GetTVShows.

### PR DESCRIPTION
**Pull request by ghost**
Adds a boolean parameter to VideoLibrary.GetTVShows to allow retrieving TV shows without episodes even if `videolibrary.hideemptyseries` is true in advancedsettings.xml, as it was applied to _everything_ and not just the library view.
Although I think that this should be the default behavior, I have added it as an option in case existing clients expect empty series to be also "hidden" from JSON-RPC.
Fixes MilhouseVH/texturecache.py#25.

There's also a fix for CEdenVideoArtUpdater as hideemptyseries was applied to it too and that is clearly wrong.

### CODE REVIEW

**Comment by anaconda** [comment]
Ping @Montellese

File: xbmc/video/VideoDatabase.cpp [hunk 1]
```diff
@@ -6220,6 +6220,8 @@ bool CVideoDatabase::GetTvShowsNav(const std::string& strBaseDir, CFileItemList&
     videoUrl.AddOption("tagid", idTag);
 
   Filter filter;
+  if (!CSettings::GetInstance().GetBool(CSettings::SETTING_VIDEOLIBRARY_SHOWEMPTYTVSHOWS))
+    filter.AppendWhere("totalCount IS NOT NULL AND totalCount>0");
```
Comment (Montellese) [inline xbmc/video/VideoDatabase.cpp]: Please add spaces around the `>`.

**Comment by Montellese** [comment]
Hm this still looks like the output of JSON-RPC's `VideoLibrary.GetTvShows` depends upon the value of the `videolibrary.showemptytvshows` or am I missing something?

**Comment by anaconda** [comment]
Updated.
@Montellese this PR updates JSON-RPC `CVideoLibrary::GetTVShows` to call `CVideoDatabase::GetTvShowsByWhere` instead of `CVideoDatabase::GetTvShowsNav`, where the setting check is moved. So library checks the setting, but JSON-RPC API does not.

**Comment by Montellese** [comment]
Ah ok, that's fine with me then.

**Comment by Razzeee** [comment]
jenkins build this please

**Comment by Tolriq** [comment]
@Montellese maybe you should make an announcement to slow JSON RPC merge until you decide how to handle API versioning now that API is 7 else it will be too late :( Too bad this was merged :(

**Pull request merged**